Sources/AppcuesKit/Presentation/Actions/ActionRegistry.swift

```diff
@@ -9,13 +9,11 @@
 import Foundation
 
 internal class ActionRegistry {
-    typealias Completion = () -> Void
-
     private var actions: [String: AppcuesExperienceAction.Type] = [:]
 
     private var isProcessing = false
     private var actionQueue: [AppcuesExperienceAction] = [] {
-        didSet { processFirstAction() }
+        didSet { Task { try await processFirstAction() } }
```

Comment:
This might be a good candidate to convert to an `AsyncSequence`. Will look into doing so later.